src/OpenApi/src/Services/Schemas/OpenApiSchemaStore.cs

```diff
@@ -59,13 +60,7 @@ internal sealed class OpenApiSchemaStore
     /// <returns>A <see cref="JsonObject" /> representing the JSON schema associated with the key.</returns>
     public JsonNode GetOrAdd(OpenApiSchemaKey key, Func<OpenApiSchemaKey, JsonNode> valueFactory)
     {
-        if (_schemas.TryGetValue(key, out var schema))
-        {
-            return schema;
-        }
-        var targetSchema = valueFactory(key);
-        _schemas.Add(key, targetSchema);
-        return targetSchema;
+        return _schemas.GetOrAdd(key, valueFactory(key));
```

Comment:
@BrennanConroy shared some feedback here that we can use a different overload of the `GetOrAdd` method here to avoid another possible race condition:

```suggestion
        return _schemas.GetOrAdd(key, valueFactory, key);
```

With this pattern, the [GetOrAdd method](https://source.dot.net/#System.Collections.Concurrent/System/Collections/Concurrent/ConcurrentDictionary.cs,1260) will do a check for the schema key in the dictionary before calling into the factory.

The current overload used will always call the `valueFactory` even if the [key already exists in the dictionary](https://source.dot.net/#System.Collections.Concurrent/System/Collections/Concurrent/ConcurrentDictionary.cs,1292).

In this case, the `valueFactory` were calling is an invocation into the `JsonSchemaMapper`. I dunno if it is particularly prone to race issues but it still doesn't hurt to avoid unnecessarily calling into it.

Reply:
Thank you for the feedback. One small PR and I already learned a lot!
Unfortunately, the suggested overload does not work because it expects a valueFactory of type  `Func<TKey, TArg, TValue>`. Luckily, there is [another overload](https://source.dot.net/#System.Collections.Concurrent/System/Collections/Concurrent/ConcurrentDictionary.cs,1220), so I could do the following:
```diff
- return _schemas.GetOrAdd(key, valueFactory(key));
+ return _schemas.GetOrAdd(key, _ => valueFactory(key));
```

Reply:
I think this should work, and will avoid needing to allocate a closure:

```csharp
return _schemas.GetOrAdd(key, valueFactory);
```